diff --git a/src/commands/query.rs b/src/commands/query.rs
index a8cd64c..811df1e 100644
--- a/src/commands/query.rs
+++ b/src/commands/query.rs
@@ -112,8 +112,11 @@ impl StateBroadcastIntent {
                     command:
                         Command::Window(
                             Operation::VirtualMove(_, _) | Operation::VirtualMoveNumber(_, _),
                         ),
                 } => intent.windows_changed = true,
+                Event::Command {
+                    command: Command::Window(Operation::Swap(_)),
+                } => intent.windows_changed = true,
                 Event::WindowFocused { .. } => intent.window_focused = true,
                 Event::WindowTitleChanged { window_id } => {
                     intent.title_changes.insert(*window_id);
